Mantle/MTLModel+NSCoding.m

```diff
@@ -128,15 +128,10 @@ - (id)decodeValueForKey:(NSString *)key withCoder:(NSCoder *)coder modelVersion:
 
 	SEL selector = MTLSelectorWithCapitalizedKeyPattern("decode", key, "WithCoder:modelVersion:");
 	if ([self respondsToSelector:selector]) {
-		NSInvocation *invocation = [NSInvocation invocationWithMethodSignature:[self methodSignatureForSelector:selector]];
-		invocation.target = self;
-		invocation.selector = selector;
-		[invocation setArgument:&coder atIndex:2];
-		[invocation setArgument:&modelVersion atIndex:3];
-		[invocation invoke];
-
-		__unsafe_unretained id result = nil;
-		[invocation getReturnValue:&result];
+		IMP imp = [self methodForSelector:selector];
+		id(*function)(id, SEL, id, NSUInteger) = (void *)imp;
```

Comment:
Can you declare the `NSCoder *` explicitly as such? It doesn't make a semantic difference, but it might clarify things to readers of the code.